src/views/Project/index.jsx

```diff
@@ -238,12 +271,42 @@ export default class Project extends Component {
     });
   };
 
-  load = async () => {
+  load = () => {
+    this.setState({ isNextPageLoading: true });
+
     const project = projects[this.props.match.params.project];
     const repositories = mergeAll(project.repositories);
     const tagsMapping = tagReposMapping(repositories);
+    const productWithComponentList = mergeAll(
+      project.products
+        ? project.products.filter(product => typeof product !== 'string')
+        : []
+    );
+    const productWithNoComponentList = project.products
+      ? project.products.filter(product => typeof product === 'string')
+      : [];
 
-    await Promise.all(
+    if (
```

Comment:
Does this `if` block mean to reset the cursor information? Is this used when user moves to different Project page and come back? If that's the case, can we set it to be empty whenever user enters this page, e.g using componentDidMount? Otherwise, feel free to add comments on this.

Reply:
You are correct. I fixed it. At first I just reused the same logic as Languages, where you can't use component did mount because the table is all that changes and not the whole component in Languages.